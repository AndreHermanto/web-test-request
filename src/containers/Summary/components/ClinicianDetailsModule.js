import React, { Component } from 'react';
import { 
  Row, 
  Col,
  Glyphicon,
  Button 
} from 'react-bootstrap';
import { 
  SummaryBox,
  SummaryHeading,
  SummaryBreakLine,
  SummaryDetails,
  SummaryTitle
} from '../summaryStyled'; 

/**
* ClinicianDetailsModule summary - UI for summary page to display all clinician details data.
* also display copy to other clinicians
*/
class ClinicianDetailsModule extends Component
{
  constructor(props) {
    super(props);
    this.handleOnClick = this.handleOnClick.bind(this);
  }
  handleOnClick() {
    this.props.handleOnClick('step5');
    return 5;
  }
  render() {
    return (
      <SummaryBox>
        <SummaryHeading> 
          Clinician - {this.props.clinicianDetailsModule.firstName + ' ' + this.props.clinicianDetailsModule.lastName}
          {this.props.handleOnClick &&
            <Button 
              bsStyle="link"
              onClick={this.handleDetailsClick}
            >
              <Glyphicon glyph="pencil"/> Edit
            </Button>
          }
        </SummaryHeading>
        <SummaryBreakLine/>
        <Row>
          <Col md={12}>
            <SummaryDetails>
              <SummaryTitle> Organisation/Address of practice:  </SummaryTitle>
              {this.props.clinicianDetailsModule.organisation}
            </SummaryDetails>
          </Col>
          {
            this.props.clinicianDetailsModule.providerNumber !== '' &&
            <Col md={12}>
              <SummaryDetails> 
                <SummaryTitle> Provider number:  </SummaryTitle>
                {this.props.clinicianDetailsModule.providerNumber}
              </SummaryDetails>
            </Col>
          }
          <Col md={12}>
            <SummaryDetails> 
              <SummaryTitle> Medical specialty:  </SummaryTitle>
              {this.props.clinicianDetailsModule.medicalSpecialty}
            </SummaryDetails>
          </Col>
          <Col md={12}>
            <SummaryDetails> 
              <SummaryTitle> Phone:  </SummaryTitle>
              {this.props.clinicianDetailsModule.phone}
            </SummaryDetails>
          </Col>
          <Col md={12}>
            <SummaryDetails> 
              <SummaryTitle> Email:  </SummaryTitle>
              {this.props.clinicianDetailsModule.email}
            </SummaryDetails>
          </Col>
          {
            this.props.clinicianDetailsModule.fax !== '' &&
            <Col md={12}>
              <SummaryDetails> 
                <SummaryTitle> Fax:  </SummaryTitle>
                {this.props.clinicianDetailsModule.fax}
              </SummaryDetails>
            </Col>
          }
        </Row>
        {
          this.props.clinicianDetailsModule.copyToHCP.length > 0 && 
          this.props.clinicianDetailsModule.copyToHCP.map((c, i) => {
            return <Row key={i}>
              <Col md={12}>
                <SummaryBreakLine/>
                <SummaryHeading> 
                  Copy of report to - {c.firstName} {c.lastName}
                </SummaryHeading>
              </Col>
              <Col md={12}>
                <SummaryDetails>
                  <SummaryTitle> Organisation/Address of practice:  </SummaryTitle>
                  {c.organisation}
                </SummaryDetails>
              </Col>
              <Col md={12}>
                <SummaryDetails> 
                  <SummaryTitle> Email: </SummaryTitle>
                  {c.email}
                </SummaryDetails>
              </Col>
            </Row>
          })
        }
      </SummaryBox>
    )
  }
}

export default ClinicianDetailsModule;