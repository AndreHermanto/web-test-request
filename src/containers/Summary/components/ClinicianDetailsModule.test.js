import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TestUtils from 'react-addons-test-utils';
import ClinicianDetailsModule from './ClinicianDetailsModule';

function getStep(value) {
  return value;
}
describe('ClinicianDetailsModule test', () => {
  const clinicianDetails = {
    "providerNumber":"as","medicalSpecialty":"as","firstName":"John",
    "lastName":"Doe","organisation":"as","phone":"as","email":"as@aca.com","fax":"as",
    "copyToHCP":[{"additionalFirstName":"asdf","additionalLastName":"asdf","additionalOrganisation":"adsf","additionalEmail":"asfd"},
    {"additionalFirstName":"asdf","additionalLastName":"asdf","additionalOrganisation":"adsf","additionalEmail":"asdf"}],
    "copy":false
  }
  const props = {
    clinicianDetailsModule: clinicianDetails,
    handleOnClick: getStep
  };
    
  test('renders without crashing ', () => {
    const page = renderer.create(React.createElement(ClinicianDetailsModule, props)).toJSON();
    expect(page).toMatchSnapshot();
  });
  
  test('handleOnClick test', () => {
    var page = TestUtils.renderIntoDocument(
      <ClinicianDetailsModule clinicianDetailsModule={clinicianDetails} handleOnClick={getStep}/>
    );
    expect(page.props.handleOnClick('step5')).toEqual('step5');
    expect(page.handleOnClick()).toEqual(5);
  });
});
