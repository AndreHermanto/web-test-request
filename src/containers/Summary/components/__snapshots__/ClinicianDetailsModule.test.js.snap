exports[`ClinicianDetailsModule test renders without crashing  1`] = `
<div
  className="cFAMBN">
  <h3
    className="gElTrd">
    Clinician - 
    John Doe
    <button
      className="btn btn-link"
      disabled={false}
      onClick={[Function]}
      type="button">
      <span
        className="glyphicon glyphicon-pencil" />
       Edit
    </button>
  </h3>
  <div
    className="dgEelt" />
  <div
    className="row">
    <div
      className="col-md-12">
      <p
        className="doDOdK">
        <span
          className="jXWIEx">
           Organisation/Address of practice:  
        </span>
        as
      </p>
    </div>
    <div
      className="col-md-12">
      <p
        className="doDOdK">
        <span
          className="jXWIEx">
           Provider number:  
        </span>
        as
      </p>
    </div>
    <div
      className="col-md-12">
      <p
        className="doDOdK">
        <span
          className="jXWIEx">
           Medical specialty:  
        </span>
        as
      </p>
    </div>
    <div
      className="col-md-12">
      <p
        className="doDOdK">
        <span
          className="jXWIEx">
           Phone:  
        </span>
        as
      </p>
    </div>
    <div
      className="col-md-12">
      <p
        className="doDOdK">
        <span
          className="jXWIEx">
           Email:  
        </span>
        as@aca.com
      </p>
    </div>
    <div
      className="col-md-12">
      <p
        className="doDOdK">
        <span
          className="jXWIEx">
           Fax:  
        </span>
        as
      </p>
    </div>
  </div>
  <div
    className="row">
    <div
      className="col-md-12">
      <div
        className="dgEelt" />
      <h3
        className="gElTrd">
        Copy of report to - 
        asdf
         
        asdf
      </h3>
    </div>
    <div
      className="col-md-12">
      <p
        className="doDOdK">
        <span
          className="jXWIEx">
           Organisation/Address of practice:  
        </span>
        adsf
      </p>
    </div>
    <div
      className="col-md-12">
      <p
        className="doDOdK">
        <span
          className="jXWIEx">
           Email: 
        </span>
        asfd
      </p>
    </div>
  </div>
  <div
    className="row">
    <div
      className="col-md-12">
      <div
        className="dgEelt" />
      <h3
        className="gElTrd">
        Copy of report to - 
        asdf
         
        asdf
      </h3>
    </div>
    <div
      className="col-md-12">
      <p
        className="doDOdK">
        <span
          className="jXWIEx">
           Organisation/Address of practice:  
        </span>
        adsf
      </p>
    </div>
    <div
      className="col-md-12">
      <p
        className="doDOdK">
        <span
          className="jXWIEx">
           Email: 
        </span>
        asdf
      </p>
    </div>
  </div>
</div>
`;
